src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.TrailingZeroCount.cs

```diff
@@ -22,13 +26,52 @@ public static void TrailingZeroCount<T>(ReadOnlySpan<T> x, Span<T> destination)
             InvokeSpanIntoSpan<T, TrailingZeroCountOperator<T>>(x, destination);
 
         /// <summary>T.TrailingZeroCount(x)</summary>
-        private readonly struct TrailingZeroCountOperator<T> : IUnaryOperator<T, T> where T : IBinaryInteger<T>
+        private readonly unsafe struct TrailingZeroCountOperator<T> : IUnaryOperator<T, T> where T : IBinaryInteger<T>
         {
-            public static bool Vectorizable => false; // TODO: Vectorize
+            public static bool Vectorizable =>
+                (AdvSimd.IsSupported && AdvSimd.Arm64.IsSupported && sizeof(T) == 1) ||
+                PopCountOperator<T>.Vectorizable; // http://0x80.pl/notesen/2023-01-31-avx512-bsf.html#trailing-zeros-simplified
+
             public static T Invoke(T x) => T.TrailingZeroCount(x);
-            public static Vector128<T> Invoke(Vector128<T> x) => throw new NotSupportedException();
-            public static Vector256<T> Invoke(Vector256<T> x) => throw new NotSupportedException();
-            public static Vector512<T> Invoke(Vector512<T> x) => throw new NotSupportedException();
+
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static Vector128<T> Invoke(Vector128<T> x)
+            {
+                if (AdvSimd.IsSupported && sizeof(T) == 1)
+                {
+                    return AdvSimd.LeadingZeroCount(AdvSimd.Arm64.ReverseElementBits(x.AsByte())).As<byte, T>();
+                }
+
+                if (PopCountOperator<T>.Vectorizable)
+                {
+                    return PopCountOperator<T>.Invoke(~x & (x - Vector128<T>.One));
+                }
+
+                throw new NotSupportedException();
+            }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static Vector256<T> Invoke(Vector256<T> x)
+            {
+                if (PopCountOperator<T>.Vectorizable)
+                {
+                    PopCountOperator<T>.Invoke(~x & (x - Vector256<T>.One));
```

Comment:
Does this compile?  Seems like it's missing a `return `.  Same below.

Reply:
It does. Seems like it was doing extra work for no good reason.